extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -190,54 +191,49 @@ export class BookTocManager implements IBookTocManager {
 	 * @param addSection The section that'll be added to the target section. If it's undefined then the target section (findSection) is removed from the table of contents.
 	*/
 	async updateTOC(version: BookVersion, tocPath: string, findSection: JupyterBookSection, addSection?: JupyterBookSection): Promise<void> {
-		const toc = yaml.safeLoad((await fs.readFile(tocPath, 'utf8')));
-		this._tocFiles.set(tocPath, toc);
-		let newToc = new Array<JupyterBookSection>(toc.length);
-		for (const [index, section] of toc.entries()) {
-			let newSection = this.buildTOC(version, section, findSection, addSection);
-			if (newSection) {
-				newToc[index] = newSection;
-			}
+		const tocFile = await fs.readFile(tocPath, 'utf8');
+		this.tableofContents = yaml.safeLoad(tocFile);
+		if (!this._tocFiles.has(tocPath)) {
```

Comment:
If tocFiles already has the path in it then you won't be updating that map - is that expected behavior? It seems like you should be overwriting it with the new one regardless. Although I'm curious if you ever expect the path to be the same anyways - would that happen during a reload or something?

Reply:
Yes, so this is specifically for moving a section within the same book. In that case, it'll remove the section and overwrite the toc.